File: packages/shared/src/types/feeding.ts
/**
 * Feeding types for horse feeding management
 */

export type FeedCategory =
  | "roughage"
  | "concentrate"
  | "supplement"
  | "medicine";

export type QuantityMeasure =
  | "scoop"
  | "teaspoon"
  | "tablespoon"
  | "cup"
  | "ml"
  | "l"
  | "g"
  | "kg"
  | "custom";

/**
 * Feed type definition (e.g., "Müsli Plus", "Hay")
 * Organization-scoped configuration - shared across all stables in the organization
 */
export interface FeedType {
  id: string;
  organizationId: string;
  name: string;
  brand?: string | null;
  category: FeedCategory;
  quantityMeasure: QuantityMeasure;
  defaultQuantity: number;
  warning?: string;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * Feeding time slot definition (e.g., "morning" at 07:00)
 * Stable-scoped configuration
 */
export interface FeedingTime {
  id: string;
  stableId: string;
  name: string;
  time: string; // HH:mm format
  sortOrder: number;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
}

/**
 * Horse feeding assignment
 * Links a horse to a feed type at a specific feeding time
 */
export interface HorseFeeding {
  id: string;
  stableId: string;
  horseId: string;
  feedTypeId: string;
  feedingTimeId: string;
  quantity: number;
  startDate: string;
  endDate?: string;
  notes?: string;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
  // Denormalized fields for display efficiency
  feedTypeName: string;
  feedTypeCategory: FeedCategory;
  quantityMeasure: QuantityMeasure;
  horseName: string;
  feedingTimeName: string;
}

// ============================================================
// Create/Update DTOs
// ============================================================

export interface CreateFeedTypeData {
  name: string;
  brand?: string | null;
  category: FeedCategory;
  quantityMeasure: QuantityMeasure;
  defaultQuantity: number;
  warning?: string;
}

export interface UpdateFeedTypeData extends Partial<CreateFeedTypeData> {
  isActive?: boolean;
}

export interface CreateFeedingTimeData {
  name: string;
  time: string;
  sortOrder?: number;
}

export interface UpdateFeedingTimeData extends Partial<CreateFeedingTimeData> {
  isActive?: boolean;
}

export interface CreateHorseFeedingData {
  horseId: string;
  feedTypeId: string;
  feedingTimeId: string;
  quantity: number;
  startDate: string;
  endDate?: string;
  notes?: string;
}

export interface UpdateHorseFeedingData extends Partial<CreateHorseFeedingData> {
  isActive?: boolean;
}

File: EquiDuty-Android/app/src/main/java/com/equiduty/domain/model/Feeding.kt
package com.equiduty.domain.model

data class FeedType(
    val id: String,
    val stableId: String,
    val name: String,
    val brand: String,
    val category: FeedCategory,
    val quantityMeasure: QuantityMeasure,
    val defaultQuantity: Double,
    val warning: String?,
    val isActive: Boolean,
    val createdBy: String,
    val createdAt: String,
    val updatedAt: String
)

data class FeedingTime(
    val id: String,
    val stableId: String,
    val name: String,
    val time: String, // HH:mm
    val sortOrder: Int,
    val isActive: Boolean,
    val createdBy: String,
    val createdAt: String,
    val updatedAt: String
)

data class HorseFeeding(
    val id: String,
    val stableId: String,
    val horseId: String,
    val feedTypeId: String,
    val feedingTimeId: String,
    val quantity: Double,
    val startDate: String,
    val endDate: String?,
    val notes: String?,
    val isActive: Boolean,
    val createdBy: String,
    val createdAt: String,
    val updatedAt: String,
    val feedTypeName: String,
    val feedTypeCategory: FeedCategory,
    val quantityMeasure: QuantityMeasure,
    val horseName: String,
    val feedingTimeName: String
) {
    val formattedQuantity: String
        get() = "${quantity.let { if (it == it.toLong().toDouble()) it.toLong().toString() else "%.1f".format(it) }} ${quantityMeasure.abbreviation}"
}

enum class FeedCategory(val value: String) {
    ROUGHAGE("roughage"), CONCENTRATE("concentrate"),
    SUPPLEMENT("supplement"), MEDICINE("medicine");

    companion object {
        fun fromValue(value: String): FeedCategory =
            entries.find { it.value == value } ?: ROUGHAGE
    }
}

enum class QuantityMeasure(val value: String, val abbreviation: String) {
    SCOOP("scoop", "sk"),
    TEASPOON("teaspoon", "tsk"),
    TABLESPOON("tablespoon", "msk"),
    CUP("cup", "cup"),
    ML("ml", "ml"),
    L("l", "l"),
    G("g", "g"),
    KG("kg", "kg"),
    CUSTOM("custom", "");

    companion object {
        fun fromValue(value: String): QuantityMeasure =
            entries.find { it.value == value } ?: KG
    }
}
